test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -697,6 +701,17 @@ var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
 				ginkgo.By("creating the replacement pod and readmitting the workload will unsuspended the replacement", func() {
 					gomega.Expect(k8sClient.Create(ctx, replacementPod)).Should(gomega.Succeed())
 
+					util.SetNodeName(ctx, k8sClient, "node1", replacementPod)
+
+					// Eventually pod2 fully removed
+					gomega.Eventually(func(g gomega.Gomega) bool {
+						err := k8sClient.Get(ctx, pod2LookupKey, createdPod)
+						if err != nil {
+							return apierrors.IsNotFound(err)
+						}
+						return createdPod.DeletionTimestamp.IsZero()
```

Comment:
we should wait for "not found" only, we have already establish that `createdPod.DeletionTimestamp` is not zero  a couple of lines above for this, making this wait useless.